### PR TITLE
Hide overlay on bank opened option

### PR DESCRIPTION
**Pull request by Obasill**
Adds option to hide quest step overlay when bank is currently open.

### CODE REVIEW

File: src/main/java/com/questhelper/QuestHelperConfig.java [hunk 1]
```diff
@@ -200,6 +200,16 @@ default boolean showOverlay()
 		return true;
 	}
 
+	@ConfigItem(
+		keyName = "hideOverlayPanelBank",
+		name = "Hide overlay when bank opened",
+		description = "Chose whether the overlay should be displayed when bank opened"
```
Comment (tivervac) [inline src/main/java/com/questhelper/QuestHelperConfig.java:206]:
```suggestion
		description = "Choose whether the overlay should be displayed when a bank tab is open"
```

File: src/main/java/com/questhelper/overlays/QuestHelperOverlay.java [hunk 2]
```diff
@@ -33,25 +33,39 @@
...
 		this.plugin = plugin;
+		setLayer(OverlayLayer.ALWAYS_ON_TOP);
 		setPriority(OverlayPriority.HIGHEST);
 	}
 
 	@Override
 	public Dimension render(Graphics2D graphics)
 	{
+		Widget itemContainer = client.getWidget(WidgetInfo.BANK_ITEM_CONTAINER);
+		if (plugin.getConfig().showOverlay() && plugin.getConfig().hideOverlayBank() && !(itemContainer == null))
```
Comment (tivervac) [inline src/main/java/com/questhelper/overlays/QuestHelperOverlay.java:65]:
```suggestion
		if (plugin.getConfig().showOverlay() && plugin.getConfig().hideOverlayBank() && itemContainer != null)
```

Comment (Obasill) [reply src/main/java/com/questhelper/overlays/QuestHelperOverlay.java:65]: Is there a reason for this change? I'm not opposed to it, I'm simply curious.

Comment (tivervac) [reply src/main/java/com/questhelper/overlays/QuestHelperOverlay.java:65]: The semantics are the same, but it’s just a tad more readable

**Comment by Obasill** [comment]
Updated branch so it's ready for merge

**Comment by Zoinkwiz** [comment]
The overlay should go behind the bank interface now, so this shouldn't be necessary now.

behindinterface

**Comment by Obasill** [comment]
Makes sense, which commit changed this? Cause I remember looking into this and didn't bother cause I found was a lot more work than this solution.

**Pull request closed without merging**